keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1165 @@
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./tests-runner"]
+```
+
+In this case, we run 10 indexes representing the test suites. We allow for one
+failure per index.
+
+#### Story 2
+
+As a CI/CD platform administrator from the [Story 1](#story-1) I want to be able
+to control the failures with the pod failure policy. In particular, I want
+to be able to use pod failure policy to avoid restarts of some indexes, based
+on exit codes.
+
+The following Job configuration could satisfy my use case:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  backoffLimit: 1
+  runPolicy:
+    backoffLimitPerIndex: true
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./tests-runner"]
+  podFailurePolicy:
+    rules:
+    - action: FailIndex
+      onExitCodes:
+        operator: In
+        values: [42]
+```
+
+#### Story 3
+
+As a CI/CD platform administrator from the [Story 1](#story-1) I want to be able
+to fail the entire Job if the number of failed indexes exceeds 50%. I want to
+do this in order to cut down costs of running the tests in case of compilation
+issues that would result in all tests failing.
+
+The following Job configuration could satisfy my use case:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  runPolicy:
+    backoffLimitPerIndex: true
+    maxFailedIndexes: 5
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./tests-runner"]
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+
+#### The Job object too big
+
+With the new field `.status.failedIndexes` the Job object can be significantly
+larger as every failed index is recorded in the field. For example, using the
+same format as for the `.status.completedIndexes` and allowing for completions
+equal to `10^6` the size of the field, in the worst-case scenario is 3.4MB, which
+could put significant overhead on fetching the Job object data from etcd.
+
+In order to mitigate this risk we constrain the maximal value of the
+`.spec.completions` field to `10^5`, when `.spec.runPolicy.backoffLimitPerIndex=true`.
+With this limit the size of the field in the worst-case scenario (of
+non-consecutive integers from `0` to `.spec.completions-1`) is limited by `295KB`.
+Even if the `.status.completedIndexes` reaches its limits, still the size of
+these two fields together will be limited by `690KB`, which is enough for etcd
```

Comment:
completedIndexes & failedIndexes. Added this into text to make clear. Also reference this section from Design Details to avoid duplication of the numbers.